hera_qm/xrfi.py

```diff
@@ -489,6 +563,7 @@ def flag_apply(uvf, uv, keep_existing=True, force_pol=False, history='',
         if f.mode != 'flag':
             raise ValueError('UVFlag objects must be in mode "flag" to apply to data.')
         if f.type == 'waterfall':
+            f = f.copy()  # don't change the input object
```

Comment:
Does this prevent the input object from being changed? I've usually seen a `deepcopy` used to prevent accidental modification, but if `copy()` does the same thing, this is fine.

Reply:
It's `UVFlag` magic! The `copy` method actually just does a deep copy underneath.